src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -4,10 +4,11 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using OSPSuite.Core.Extensions;
 
 namespace OSPSuite.Core.Domain.Services
 {
-   public class ConcurrencyManagerResult<TResult> where TResult : class
+   public class ConcurrencyManagerResult<TResult>
```

Comment:
@abdelr Not sure why this constraint was required. I removed it as I was writing a test